packages/material-ui-lab/src/TimePicker/TimePickerToolbar.tsx

```diff
@@ -9,61 +14,120 @@ import { useUtils } from '../internal/pickers/hooks/useUtils';
+export interface TimePickerToolbarProps extends ToolbarComponentProps {
+  classes?: {
+    separator?: string;
+    hourMinuteLabel?: string;
+    hourMinuteLabelLandscape?: string;
+    hourMinuteLabelReverse?: string;
+    ampmSelection?: string;
+    ampmLandscape?: string;
+    ampmLabel?: string;
+    penIconLandscape?: string;
+  };
+}
+
+export type TimePickerToolbarClassKey = keyof NonNullable<TimePickerToolbarProps['classes']>;
+
+export function getTimePickerToolbarUtilityClass(slot: string) {
+  return generateUtilityClass('PrivateTimePickerToolbar', slot);
+}
+
+export const timePickerToolbarClasses = generateUtilityClasses<TimePickerToolbarClassKey>(
+  'PrivateTimePickerToolbar',
+  [
+    'separator',
+    'hourMinuteLabel',
+    'hourMinuteLabelLandscape',
+    'hourMinuteLabelReverse',
+    'ampmSelection',
+    'ampmLandscape',
+    'ampmLabel',
+    'penIconLandscape',
+  ],
+);
+
+const useUtilityClasses = (styleProps: TimePickerToolbarProps & { theme: Theme }) => {
+  const { theme, isLandscape, classes } = styleProps;
+
+  const slots = {
+    penIconLandscape: ['penIconLandscape'],
+    separator: ['separator'],
+    hourMinuteLabel: [
+      'hourMinuteLabel',
+      isLandscape && 'hourMinuteLabelLandscape',
+      theme.direction === 'rtl' && 'hourMinuteLabelReverse',
+    ],
+    ampmSelection: ['ampmSelection', isLandscape && 'ampmLandscape'],
+    ampmLabel: ['ampmLabel'],
+  };
+
+  return composeClasses(slots, getTimePickerToolbarUtilityClass, classes);
+};
+
+const TimePickerToolbarRoot = styled(
+  PickersToolbar,
+  {},
+  { skipSx: true },
+)({
+  [`& .${timePickerToolbarClasses.penIconLandscape}`]: {
     marginTop: 'auto',
   },
-  hourMinuteLabelReverse: {
+});
+
+const TimePickerToolbarSeparator = styled(
+  PickersToolbarText,
+  {},
+  { skipSx: true },
+)({
+  outline: 0,
+  margin: '0 4px 0 2px',
+  cursor: 'default',
+});
+
+const TimePickerToolbarHourMinuteLabel = styled(
+  'div',
+  {},
+  { skipSx: true },
+)(({ theme, styleProps = {} }) => ({
+  display: 'flex',
+  justifyContent: 'flex-end',
+  alignItems: 'flex-end',
+  ...(!!styleProps.isLandscape && {
+    marginTop: 'auto',
+  }),
+  ...(theme.direction === 'rtl' && {
     flexDirection: 'row-reverse',
-  },
-  ampmSelection: {
-    display: 'flex',
-    flexDirection: 'column',
-    marginRight: 'auto',
-    marginLeft: 12,
-  },
-  ampmLandscape: {
+  }),
+}));
+
+const TimePickerToolbarAmPmSelection = styled(
+  'div',
+  {},
+  { skipSx: true },
+)(({ styleProps = {} }) => ({
+  display: 'flex',
+  flexDirection: 'column',
+  marginRight: 'auto',
+  marginLeft: 12,
+  ...(!!styleProps.isLandscape && {
     margin: '4px 0 auto',
     flexDirection: 'row',
     justifyContent: 'space-around',
     flexBasis: '100%',
-  },
-  ampmLabel: {
+  }),
+  [`& .${timePickerToolbarClasses.ampmLabel}`]: {
     fontSize: 17,
   },
-  penIconLandscape: {
-    marginTop: 'auto',
-  },
-};
+}));
 
 /**
  * @ignore - internal component.
  */
```

Comment:
@mnajdova this part indicate as internal component

Reply:
Fair enough, have missed it. I think we should remove it from the `overrides.d.ts` May have been missed in @eps1lon's PR, but let's wait on his response, in case it was intentional